### PR TITLE
Feature: create migration 7.28.0

### PR DESCRIPTION
**Pull request by matheushahnn**
## Context

Add liquidities and tokens for Rootstock
Also add some small improvements to avoid running invalid liquidities job (when missing subgraph URL)

## To-do list

- [ ] Added tests
- [ ] Tested on dev/sbx
- [ ] Changelog was updated with current changes
- [ ] Documentation or guides were updated (slab, readme)
- [ ] Version was updated on `package.json` (releases/hotfixes)

## Checklist before merge

- [ ] **there are no errors in logs in any workers**
- [ ] new blocks are being discovered and marked as finalized
- [ ] foreign blocks are being dispatched
- [ ] squash all commits before merging

### CODE REVIEW

File: src/services/Migrations.ts [hunk 1]
```diff
@@ -40,6 +45,151 @@ class Migrations {
       console.log(`collection ${CachedValidator} removed.`);
     });
   };
+
+  private static migrateTo7280 = async () => {
+    await Migrations.wrapMigration('7.28.0', async () => {
```
Comment (matheushahnn) [inline src/services/Migrations.ts:50]: Is the correct version number?

File: src/services/Migrations.ts [hunk 2]
```diff
@@ -40,6 +45,151 @@ class Migrations {
...
+async function getUniswapV3Migration(uniswapV3PoolModel: ModelType<UniswapV3Pool>) {
+  const uniswapV3Pools = [
+    {
+      chainId: 'rootstock',
+      protocol: 'uniswapV3',
+      token0: '0x542fDA317318eBF1d3DEAf76E0b632741A7e677d',
+      token1: '0xef213441A85dF4d7ACbDaE0Cf78004e1E486bB96',
+      fee: 500,
+      address: '0xd2ffe51ab4e622a411abbe634832a19d919e9c55',
+      liquidityActive: '17716643053084368348144',
```
Comment (matheushahnn) [inline src/services/Migrations.ts:82]: liquidityActive, liquidityLockedToken0, and liquidityLockedToken1 are hardcoded values and do not represent the value in the pool.

File: src/services/Migrations.ts (hunk 1 above)
Comment (DZariusz) [reply src/services/Migrations.ts:50]: yes, is is ok.

File: src/services/Migrations.ts [hunk 3]
```diff
@@ -40,6 +45,151 @@ class Migrations {
       console.log(`collection ${CachedValidator} removed.`);
     });
   };
+
+  private static migrateTo7280 = async () => {
+    await Migrations.wrapMigration('7.28.0', async () => {
+      // we have different index and validator per network, so let's simply drop and rebuild
```
Comment (DZariusz) [inline src/services/Migrations.ts:51]: is this comment up to date? I do not see you droping anything

File: src/workers/UniswapV3LiquidityWorker.ts [hunk 4]
```diff
@@ -18,17 +20,25 @@ class UniswapV3LiquidityWorker extends BasicWorker {
...
 
     if (!unlocked) {
-      this.logger.error('[UniswapV3LiquidityWorker] apply for job but job !unlocked');
+      this.logger.error(`${loggerPrefix} apply for job but job !unlocked`);
+      return;
+    }
+
+    const isValidSettings = this.checkIsValidSettings(job.data);
+
+    if (!isValidSettings) {
```
Comment (DZariusz) [inline src/workers/UniswapV3LiquidityWorker.ts:35]: this check should be first one and should not display an error - you can log.debug

**Review by DZariusz — APPROVED** [review]

**Comment by DZariusz** [comment]
since you changing some stuff for logs:

about `this.logger.error('[UniswapV3MultiFetcher] no data to fetch');` - please first check if we have any input, if not log.debug this information, and only then if we have input and no pools, we can display this message  + add TODO to display for which input pools are missing (unless this is easy check then please include as well)

File: src/services/Migrations.ts (hunk 3 above)
Comment (matheushahnn) [reply src/services/Migrations.ts]: Removed

**Pull request merged**